src/ansi-c/c_typecheck_base.cpp

```diff
@@ -744,6 +760,48 @@ void c_typecheck_baset::typecheck_declaration(
           if(return_type.id() != ID_empty)
             parameter_map.erase(CPROVER_PREFIX "return_value");
         }
+
+        // create a contract symbol
+        symbolt contract;
+        contract.name = "contract::" + id2string(new_symbol.name);
+        contract.base_name = new_symbol.name;
+        contract.pretty_name = new_symbol.name;
+        contract.is_property = true;
+        contract.type = code_type;
```

Comment:
Will the contract symbol also contain the function body if one was given by the user ? 

If possible I would like the contract symbol to have an empty body but to keep the contract parameters in the symbol table, because in a second step (goto-convert maybe ?) I would like to generate an executable body from the contract clauses.

Reply:
No, the contract symbol _at present_ really just has the type information including the contract annotation. We could obviously evolve that.

I'm not sure whether symbols for the contract parameters are required to remain in the symbol table: even the type signature already has the parameters with their names and their types, so we can easily create symbols from them if need be.

Reply:
Use a binding instead!

Reply:
> Use a binding instead!

For all that I can tell this requires some preparing work. https://github.com/diffblue/cbmc/pull/6811 will be a starting point.

Reply:
I have now added a commit that wraps contracts in a `lambda_exprt` and uses `.application()` instead of resorting to `replace_symbolt`.